superduperdb/db/mongodb/query.py

```diff
@@ -848,7 +849,7 @@ def __call__(self, db: DB):
             **self.kwargs,
         )
         graph = None
-        if self.refresh and not s.CFG.cluster.cdc:
+        if self.refresh and not CDC_COLLECTION_LOCKS[self.collection.name]:
```

Comment:
You don't know if this has been set yet or not, so it has to be:

`if not CDC_COLLECTION_LOCKS.get(self.collection.name):`

Reply:
im using defaultdict

Reply:
Oh, I strongly advise against doing this for global dictionaries!

The reason is that simply _looking_ at the dictionary can change its value and break things.

Suppose you pass this some day to a function that says:

```
try:
    return d[name]
except KeyError:
  pass
```

It will add an empty value to the dictionary.

----

I'm against `defaultdict` nearly always, and that's because you get a thing that looks like a `dict` and can get passed around like a `dict` but isn't a `dict` in a very important way.

I see code snippets like `collection[key]` hundreds of times every day.  Nearly always, this snippet raises an exception if `key` is not in `collection`.  If this is not true, I have to stop and think about it, which wastes my time, but sometimes I'll miss that, and perhaps make a mistake.

Life is too short for `defaultdict`.  It makes it a tiny bit easier for the person writing the code, but significantly harder for the reader, and there are a lot more readers than writers.

:-D


`CDC_COLLECTION_LOCKS[self.collection.name]`

Reply:
what do u suggest if not defaultdict

Reply:
I used plain dict